contrib/epee/include/net/net_utils_base.h

```diff
@@ -46,14 +49,111 @@ namespace epee
 {
 namespace net_utils
 {
+	struct network_address_base
+	{
+	public:
+		bool operator==(const network_address_base &other) const { return m_full_id == other.m_full_id; }
+		bool operator!=(const network_address_base &other) const { return !operator==(other); }
+		bool operator<(const network_address_base &other) const { return m_full_id < other.m_full_id; }
+		bool is_same_host(const network_address_base &other) const { return m_host_id == other.m_host_id; }
+		virtual std::string str() const = 0;
+		virtual std::string host_str() const = 0;
+		virtual bool is_loopback() const = 0;
+		virtual bool is_local() const = 0;
+		virtual uint8_t get_type_id() const = 0;
+	protected:
+		// A very simple non cryptographic hash function by Fowler, Noll, Vo
+		uint64_t fnv1a(const uint8_t *data, size_t len) const {
+			uint64_t h = 0xcbf29ce484222325;
+			while (len--)
```

Comment:
What is the purpose of this function? If its not cryptographic, it is unlikely that it should be used for equivalence checks. Was this supposed to be use for hash tables? So that network addresses could be used as keys? I don't see that anywhere ... ?